### PR TITLE
Implement Reboot command for chip-tool-darwin.

### LINKED ISSUE #16514: Add support to reboot apps in chip-tool-darwin.
#### Problem
Currently there is no support to reboot apps in chip-tool-darwin. 

 #### Proposed Solution
Use current infrastructure to support rebooting a device in chip-tool-darwin. Add tests back to chip-tool-darwin that are set to manual that require manual interaction of reboot. This [PR](https://github.com/project-chip/connectedhomeip/pull/16039) adds reboot to TestModeSelectCluster which will need to be added back to CI tests in chip-tool-darwin when reboot is enabled.

### PR DESCRIPTION
**Pull request by bzbarsky-apple**
Enables a few tests that needed Reboot to be enabled.

Aligns the general test list better with chip-tool, so it's easier to
see what's still missing/disabled.

Fixes https://github.com/project-chip/connectedhomeip/issues/16514

#### Problem
See above.

#### Change overview
See above.

#### Testing
Should all be verified by CI, but I did run the various tests I am enabling locally, I believe.

### CODE REVIEW

File: examples/chip-tool-darwin/templates/tests/tests.js [hunk 1]
```diff
@@ -174,12 +175,21 @@ function getTests() {
     'Test_TC_MOD_1_1',
   ];
 
+  const MultipleFabrics = [
+    // These tests all need PairWithQRCode
```
Comment (krypton36) [inline examples/chip-tool-darwin/templates/tests/tests.js:179]: nit: Can this be made as a `TODO:`?

Comment (krypton36) [reply examples/chip-tool-darwin/templates/tests/tests.js:179]: I created an issue: https://github.com/project-chip/connectedhomeip/issues/17857

**Review by krypton36 — APPROVED** [review]

File: examples/chip-tool-darwin/templates/tests/tests.js (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/chip-tool-darwin/templates/tests/tests.js]: Made all the comments TODO.

**Comment by bzbarsky-apple** [comment]
Fast-tracking platform-specific test additions.

**Pull request merged**